pkg/engine/jmespath/functions.go

```diff
@@ -965,3 +980,35 @@ func jpRandom(arguments []interface{}) (interface{}, error) {
 	}
 	return ans, nil
 }
+
+func jpX509Decode(arguments []interface{}) (interface{}, error) {
+	input, err := validateArg(x509_decode, arguments, 0, reflect.String)
+	if err != nil {
+		return nil, err
+	}
+	res := make(map[string]interface{})
+	p, _ := pem.Decode([]byte(input.String()))
+	if p == nil {
+		return res, errors.New("invalid certificate")
+	}
+
+	var v interface{}
+	cert, err := x509.ParseCertificate(p.Bytes)
+	if err != nil {
+		return res, errors.WithStack(err)
+	}
+	v = struct{ *x509.Certificate }{cert}
+
+	buf := new(bytes.Buffer)
+	enc := json.NewEncoder(buf)
+	err = enc.Encode(v)
+	if err != nil {
+		return res, errors.WithStack(err)
```

Comment:
You should be able to use fmt.Errorf to provide a good error message along with the stack using `%w` here.